src/nvim/ex_docmd.c

```diff
@@ -1478,6 +1478,45 @@ bool is_cmd_ni(cmdidx_T cmdidx)
                                      || cmdnames[cmdidx].cmd_func == ex_script_ni);
 }
 
+// Find the command name after skipping range specifiers.
+static char *find_command_after_range(exarg_T *eap)
+{
+  char *cmd = eap->cmd;
+  eap->cmd = skip_range(eap->cmd, NULL);
+  if (*eap->cmd == '*') {
+    eap->cmd = skipwhite(eap->cmd + 1);
+  }
+  char *p = find_ex_command(eap, NULL);
+  eap->cmd = cmd;  // Restore original position for address parsing
+  return p;
+}
+
+// Set the forceit flag based on the presence of '!' after the command.
+static void set_forceit_flag(exarg_T *eap, char **p)
+{
+  if (**p == '!' && eap->cmdidx != CMD_substitute
+      && eap->cmdidx != CMD_smagic && eap->cmdidx != CMD_snomagic) {
+    (*p)++;
+    eap->forceit = true;
+  } else {
+    eap->forceit = false;
+  }
+}
+
+// Parse register designation and count values for the command.
+static bool parse_register_and_count(exarg_T *eap, const char **errormsg, bool for_execution)
+{
+  parse_register(eap);
+  if (parse_count(eap, errormsg, for_execution) == FAIL) {
+    return false;
+  }
+  // Clean up next command pointer
+  if (eap->nextcmd) {
+    eap->nextcmd = skip_colon_white(eap->nextcmd, true);
+  }
```

Comment:
This is specific to the `nvim_parse_cmd` API. `do_one_cmd()` already sets a different `nextcmd`.